libs/langchain/langchain/prompts/__init__.py

```diff
@@ -53,8 +52,29 @@
     load_prompt,
 )
 
+from langchain._api import create_importer
 from langchain.prompts.prompt import Prompt
 
+if TYPE_CHECKING:
+    from langchain_community.example_selectors.ngram_overlap import (
+        NGramOverlapExampleSelector,
+    )
+
+# Create a way to dynamically look up deprecated imports.
+# Used to consolidate logic for raising deprecation warnings and
+# handling optional imports.
+MODULE_LOOKUP = {
```

Comment:
ooc any reason not to make this private _MODULE_LOOKUP

Reply:
Ah no reason, just inertia -- we can follow up with another refactor to make this private